File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/count-subarrays-where-max-element-appears-at-least-k-times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        mx = max(nums)
        
        i, j, count, ans = 0, 0, 0, 0
        
        while(j < n):
            if(nums[j] == mx):
                count += 1
            if(count >= k):
                while(count >= k):
                    ans += (n - j)
                
                    if(nums[i] == mx):
                        count -= 1
                    i += 1
                    
            j += 1
            
        return ans

File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/count-subarrays-where-max-element-appears-at-least-k-times.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countSubarrays = function(nums, k)
{
    let n = nums.length;
    let mx = Math.max(...nums);

    let i = 0, j = 0, count = 0, ans = 0;

    while(j < n)
    {
        if(nums[j] == mx)
        {
            count++;
        }

        if(count >= k)
        {
            while(count >= k)
            {
                ans += (n - j);

                if(nums[i] == mx)
                {
                    count--;
                }
                i++;
            }
        }
        j++;
    }

    return ans;
};